osutil/epoll/epoll.go

```diff
@@ -0,0 +1,159 @@
+// Package epoll contains a thin wrapper around the epoll(7) facility.
+//
+// Using epoll from Go is unusual as the language provides facilities that
+// normally make using it directly pointless. Epoll is strictly required for
+// unusual kernel interfaces that use event notification but don't implement
+// file descriptors that provide usual read/write semantics.
+//
+// It might be possible to remove this code and use internal/poll and
+// syscall.RawConn but the necessary interfaces require golang 1.12 that snapd
+// cannot yet depend on.
+package epoll
+
+import (
+	"runtime"
+	"strings"
+	"syscall"
```

Comment:
should/could we use x/sys/unix instead?

Reply:
What would be the benefit of using `x/sys/unix` over `syscall`? It looks to me like they provide the same functions.

Reply:
syscall is deprecated and locked down, which means is not completely recommended for new code that uses non-trivial syscalls. See Overview here https://pkg.go.dev/syscall

Reply:
I've replaced syscall with x/sys/unix. Thanks!